contrib/babelfishpg_tsql/sql/sys_procedures.sql

```diff
@@ -271,3 +271,223 @@ GRANT EXECUTE on PROCEDURE sys.sp_babelfish_volatility(IN sys.varchar, IN sys.va
+    RAISE EXCEPTION 'An invalid parameter or option was specified for procedure ''%''.', procedure_name;
+    RETURN;
+  END IF;
+  IF "@level1type" IS NOT NULL THEN
+    IF "@level0type" IS NULL THEN
+      RAISE EXCEPTION 'An invalid parameter or option was specified for procedure ''%''.', procedure_name;
+      RETURN;
+    END IF;
+  END IF;
+  IF "@level2type" IS NOT NULL THEN
+    IF "@level0type" IS NULL OR "@level1type" IS NULL THEN
+      RAISE EXCEPTION 'An invalid parameter or option was specified for procedure ''%''.', procedure_name;
+      RETURN;
+    END IF;
+  END IF;
+
+  -- DATABASE
+  IF "@level0type" IS NULL THEN
+    IF "@level1type" IS NOT NULL OR "@level2type" IS NOT NULL THEN
+      RAISE EXCEPTION 'An invalid parameter or option was specified for procedure ''%''.', procedure_name;
+      RETURN;
+    END IF;
+
+    var_type := 'DATABASE';
+  END IF;
+
+  -- SCHEMA or object in SCHEMA
+  IF "@level0type" IN ('SCHEMA') THEN
+    IF "@level1type" IS NOT NULL AND "@level1type" NOT IN ('TABLE', 'VIEW', 'SEQUENCE', 'PROCEDURE', 'FUNCTION', 'TYPE') THEN
+      RAISE EXCEPTION 'Extended properties for object type % are not currently supported by Babelfish.', "@level1type";
+      RETURN;
+    END IF;
+    IF "@level2type" IS NOT NULL THEN
+      IF NOT ("@level1type" IN ('TABLE') AND "@level2type" IN ('COLUMN')) THEN
+        RAISE EXCEPTION 'Extended properties for object type % are not currently supported by Babelfish.', "@level2type";
+        RETURN;
+      END IF;
+    END IF;
+
+    -- validate SCHEMA
+    var_object_name := LOWER("@level0name");
+    var_schema_name := var_object_name;
+    IF sys.schema_id(var_object_name::sys.sysname) IS NULL THEN
+      RAISE EXCEPTION 'Object is invalid. Extended properties are not permitted on ''%'', or the object does not exist.', coalesce(var_object_name, 'object specified');
+      RETURN;
+    END IF;
+
+    -- SCHEMA
+    IF "@level1type" IS NULL THEN
+      var_type := 'SCHEMA';
+      var_major_name := var_schema_name;
+
+    -- object in SCHEMA
+    ELSE
+      var_major_name := LOWER("@level1name");
+
+      -- validate object in SCHEMA
+      var_object_name := var_schema_name || '.' || var_major_name;
+      IF "@level1type" = 'TABLE' THEN
+        var_object_type := ARRAY['IT', 'S', 'U'];
+      ELSIF "@level1type" = 'VIEW' THEN
+        var_object_type := ARRAY['V'];
+      ELSIF "@level1type" = 'SEQUENCE' THEN
+        var_object_type := ARRAY['SO'];
+      ELSIF "@level1type" = 'PROCEDURE' THEN
+        var_object_type := ARRAY['P', 'PC', 'RF', 'X'];
+      ELSIF "@level1type" = 'FUNCTION' THEN
+        var_object_type := ARRAY['AF', 'FN', 'FS', 'FT', 'IF', 'TF'];
+      ELSIF "@level1type" = 'TYPE' THEN
+        var_object_type := ARRAY['TT'];
+      END IF;
+      var_object_id := sys.object_id(var_object_name);
+      IF var_object_id IS NULL OR NOT EXISTS (SELECT * FROM sys.objects WHERE object_id = var_object_id AND type = ANY(var_object_type)) THEN
```

Comment:
be aware that sys.objects is currently a slow view. 
It may cause your feature slow when there are large amount of objects. 
Given the amount of call we are seeing, I suspect, this approach would not work for big migration project. 
As an alternative, if you know the object type, you could search dedicated view

Reply:
I see. Let's not workaround this bug. 
Let me file a JIRA internally and you could implement your feature assuming everything is correct